include/taskolib/GitObjectWrapper.h

```diff
@@ -0,0 +1,167 @@
+    LibGitPointer& operator=(const LibGitPointer&) = delete;
+
+    /**
+     * Function to free member variables of GitRepository object before deconstruction of their owner.
+     * necessary because they have to be freed before the libgit2 environment deconstruction, see:
+     * git_libgit2_shutdown()
+     */
+    void reset()
+    {
+        free_libgit_ptr(val_);
+        val_ = nullptr;
+    }
+
+    /// Return a non-owning pointer to the managed object (or nullptr if no object is managed).
+    T* get() {return val_;} ;
+    /// Return a constant non-owning pointer to the managed object (or nullptr if no object is managed).
+    const T* get() const {return val_;};
+
+private:
+    T *val_ = nullptr;
+};
+
+
+/**
+ * Function wrappers for libgit2 functions which manipulates pointers.
+ * (Which take double pointers as parameters)
+ * \note for documentation see the associated libgit2 function
+ *       name: "git_FUNCTIONNAME"
+ * \defgroup lgptrfunc Group of wrapped libgit2 functions
+ * \{
+ */
+
+/**
+ * Return an existing repository.
+ * \param repo_path absolute or relative path from executable
+ * \return Wrapper of a git_repository*
+*/
+LibGitPointer<git_repository> repository_open   (const std::string& repo_path);
+
+
+/**
+ * Return a fresh initialized repository.
+ * \param repo_path absolute or relative path from executable
+ * \return Wrapper of a git_repository*
+*/
+LibGitPointer<git_repository> repository_init   (const std::string& repo_path, bool is_bare);
+
+/**
+ * Return the current index of a repository.
+ * \param repo C-type repository
+ * \return Wrapper of a git_index*
+*/
+LibGitPointer<git_index>      repository_index  (git_repository* repo);
+
+/**
+ * Generate a signature from system values (already collected by the C-type repository).
+ * \param repo_path absolute or relative path from executable
+ * \return Wrapper of a git_signature*
+*/
+LibGitPointer<git_signature>  signature_default (git_repository* repo);
+
+/**
+ * Generate a signature from given parameters.
+ * \param name User name who creates the commit
+ * \param email E-mail of the user
+ * \param time timestamp
+ * \param offset timezone adjustment for the timezone
+ * \return Wrapper of a git_signature*
+*/
+LibGitPointer<git_signature>  signature_new     (const std::string& name, const std::string& email,
+                                                  int time, int offset);
+
+
+/**
+ * Collect the current tree of the git repository.
+ * \param repo C-type repository
+ * \param tree_id identity number of the active tree
+ * \return Wrapper of a git_signature*
+*/
+LibGitPointer<git_tree>       tree_lookup       (git_repository* repo, git_oid tree_id);
+
+/**
+ * Create a new status list which contains status_list_entry* elements
+ * \param repo C-type repository
+ * \param status_opt struct of status options
+*/
+LibGitPointer<git_status_list> status_list_new  (git_repository* repo, const git_status_options& status_opt);
```

Comment:
Not terribly important, but the style guide says to keep a line to no more than 90 characters.